server/src/main/java/org/opensearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -122,17 +124,21 @@ public static class Actions {
 
     private final ReplicationCollection<RecoveryTarget> onGoingRecoveries;
 
+    private final SegmentReplicationTargetService segmentReplicationTargetService;
```

Comment:
Would prefer we didn't have these services coupled.  Though I understand moving the req handler into SegmentReplicationTargetServices would require some refactoring in RecoverySourceHandler to depend on a RemoteRecoveryTargetHandler (or renamed client component) directly.  Maybe not so bad?